hoomd/hpmc/field.py

```diff
@@ -32,208 +37,136 @@ def __init__(self):
 
-    Args:
-        mc (:py:mod:`hoomd.hpmc.integrate`): MC integrator.
-        position (list): list of positions to restrain each particle (distance units).
-        orientation (list): list of orientations to restrain each particle (quaternions).
-        k (float): translational spring constant.
-        q (float): rotational spring constant.
-        symmetry (list): list of equivalent quaternions for the shape.
-        composite (bool): Set this to True when this field is part of a :py:class:`external_field_composite`.
-
-    :py:class:`lattice_field` specifies that a harmonic spring is added to every particle:
-
-    .. math::
-
-        V_{i}(r)  = k_r*(r_i-r_{oi})^2 \\
-        V_{i}(q)  = k_q*(q_i-q_{oi})^2
+    Provides common methods for all subclasses.
 
     Note:
-        1/2 is not included in the formulas, specify your spring constants accordingly.
+        Users should use the subclasses and not instantiate `ExternalField`
+        directly.
+    """
 
-    * :math:`k_r` - translational spring constant.
-    * :math:`r_{o}` - lattice positions (in distance units).
-    * :math:`k_q` - rotational spring constant.
-    * :math:`q_{o}` - lattice orientations (quaternion)
 
-    Once initialized, the compute provides the following log quantities that can be logged via analyze.log:
+class LatticeField(ExternalField):
+    R"""Restrain particles on a lattice
 
-    * **lattice_energy** -- total lattice energy
-    * **lattice_energy_pp_avg** -- average lattice energy per particle multiplied by the spring constant
-    * **lattice_energy_pp_sigma** -- standard deviation of the lattice energy per particle multiplied by the spring constant
-    * **lattice_translational_spring_constant** -- translational spring constant
-    * **lattice_rotational_spring_constant** -- rotational spring constant
-    * **lattice_num_samples** -- number of samples used to compute the average and standard deviation
+    Args:
+        reference_positions ((*N_particles*, 3) `numpy.ndarray` of
+            `float`): the reference positions of the
+            lattice :math:`[\mathrm{length}]`.
+        reference_orientations ((*N_particles*, 4) `numpy.ndarray` of
+            `float`): the reference orientations of the lattice.
+        k_translational (`float`): translational spring constant.
+        k_rotational (`float`): rotational spring constant.
+        symmetries ((*N_sym*, 4) `numpy.ndarray` of
+            `float`): the orientations that are equivalent through symmetry,
+            i.e., the rotation quaternions that leave the particles unchanged.
+
+    :py:class:`LatticeField` specifies that a harmonic spring is added to every
+    particle:
 
-    .. warning::
-        The lattice energies and standard deviations logged by this class are multiplied by the spring constant.
+    .. math::
 
-    Example::
+        V_{i}(r_i)  = k_{trans} \cdot (r_i-r_{0,i})^2 \\
+        V_{i}(q_i)  = k_{rot} \cdot (q_i-q_{0,i})^2
```

Comment:
Add a more complete description of how this is computed with respect to the given symmetries.

Reply:
Done.